test/unit/api.test.js

```diff
@@ -0,0 +1,203 @@
+'use strict';
+
+const sinon = require('sinon');
+const { expect } = require('chai');
+
+const mongodbDriver = require('mongodb');
+const mongodbLegacy = require('../..');
+const { MongoDBNamespace } = require('mongodb/lib/utils');
+const { asyncApi } = require('../tools/api');
+
+// A map of class names to their list of method details
+const classesToMethods = new Map(
+  asyncApi.map((api, _, array) => [
+    api.className,
+    new Set(array.filter(v => v.className === api.className))
+  ])
+);
+
+// Dummy data to help with testing
+const iLoveJs = 'mongodb://iLoveJavascript';
+const client = new mongodbLegacy.MongoClient(iLoveJs);
+const db = new mongodbLegacy.Db(client, 'animals');
+const collection = new mongodbLegacy.Collection(db, 'pets');
+const namespace = MongoDBNamespace.fromString('animals.pets');
+
+// A map to helpers that can create instances of the overridden classes for testing
+const OVERRIDDEN_CLASSES_GETTER = new Map([
+  ['Admin', () => new mongodbLegacy.Admin(db)],
+  ['FindCursor', () => new mongodbLegacy.FindCursor(client, namespace)],
+  ['ListCollectionsCursor', () => new mongodbLegacy.ListCollectionsCursor(db, {})],
+  ['ListIndexesCursor', () => new mongodbLegacy.ListIndexesCursor(collection)],
+  ['AggregationCursor', () => new mongodbLegacy.AggregationCursor(client, namespace)],
+  ['ChangeStream', () => new mongodbLegacy.ChangeStream(client)],
+  ['GridFSBucket', () => new mongodbLegacy.GridFSBucket(db)],
+  ['Collection', () => new mongodbLegacy.Collection(db, 'pets')],
+  ['Db', () => new mongodbLegacy.Db(client, 'animals')],
+  ['MongoClient', () => new mongodbLegacy.MongoClient(iLoveJs)]
+]);
+
+function* generateTests() {
```

Comment:
```suggestion
/**
 * A generator that yields all programatically-testable methods from the mongodb driver.  We do this in two steps:
 * First, we load the exhaustive list of all methods we need to test from api.js.  Second, we load the legacy driver 
 * and use the methods loaded from api.js to find references to each class and method we need to test.  The 
 * generator can then yield the references to the legacy classes, so that we can programatically test them.  The 
 * generator also yields all possible callback positions for each function.
 */
function* generateTests() {
```